File: epi_judge_java/epi/FirstMissingPositiveEntry.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.List;

public class FirstMissingPositiveEntry {

    @EpiTest(testDataFile = "first_missing_positive_entry.tsv")
    public static int findFirstMissingPositive(List<Integer> nums) {
        nums.sort(null);
        int i = 1;
        for (int j = 0; j < nums.size(); j++) {
            int num = nums.get(j);
            if (j > 0 && num == nums.get(j - 1)) continue;
            if (num > 0) {
                if (num != i) return i;
                i++;
            }
        }
        return i;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "FirstMissingPositiveEntry.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/first_missing_positive_entry.cc
#include <vector>

#include "test_framework/generic_test.h"
using namespace std;

// A is passed by value argument, since we change it.
int FindFirstMissingPositive(vector<int> nums) {
    set<int> s(nums.begin(), nums.end());
    for (int num = 1; num < s.size() + 2; num++) {
        if (!s.count(num)) return num;
    }
    return -1;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"A"};
    return GenericTestMain(args, "first_missing_positive_entry.cc",
                           "first_missing_positive_entry.tsv",
                           &FindFirstMissingPositive, DefaultComparator{},
                           param_names);
}
